grails-app/controllers/org/pih/warehouse/api/OrderApiController.groovy

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) 2012 Partners In Health.  All rights reserved.
+ * The use and distribution terms for this software are covered by the
+ * Eclipse Public License 1.0 (http://opensource.org/licenses/eclipse-1.0.php)
+ * which can be found in the file epl-v10.html at the root of this distribution.
+ * By using this software in any fashion, you are agreeing to be bound by
+ * the terms of this license.
+ * You must not remove this notice, or any other, from this software.
+ **/
+package org.pih.warehouse.api
+
+import grails.converters.JSON
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.importer.CSVUtils
+import org.pih.warehouse.order.Order
+import org.pih.warehouse.order.OrderItemStatusCode
+import org.pih.warehouse.order.OrderStatus
+
+import java.math.RoundingMode
+
+class OrderApiController extends BaseDomainApiController {
+
+    def orderService
+
+    def list = {
```

Comment:
@jmiranda Initially, this was meant only for purchase orders, but I realized we currently can use the purchase orders list for looking for other types of orders too. I didn't want to lose this option after the PO list UI redesign, hence I left the option to look for other order types in this endpoint too. It is modeled based on the current `OrderController.list` action, but I took out the code related to downloading order and order items CSV into separate endpoints. For Purchase Orders, results are based on the OrderSummary (we might need to think a bit more about the materialized view before the next release), and for other Order types, it's based on the Order.